pkg/monitor/cluster/summary.go

```diff
@@ -0,0 +1,52 @@
+package cluster
+
+// Copyright (c) Microsoft Corporation.
+// Licensed under the Apache License 2.0.
+
+import (
+	"context"
+	"strconv"
+)
+
+const (
+	masterRoleLabel = "node-role.kubernetes.io/master"
+	workerRoleLabel = "node-role.kubernetes.io/worker"
+)
+
+// emitSummaryMetric emits joined metric to be able to report better on all
+// clusters state in single dashboard
+func (mon *Monitor) emitSummaryMetric(ctx context.Context) error {
+	if !mon.hourlyRun {
+		return nil
+	}
+
+	cv, err := mon.getClusterVersion()
+	if err != nil {
+		return err
+	}
+
+	ns, err := mon.listNodes()
+	if err != nil {
+		return err
+	}
+
+	var masterCount, workerCount int
+	for _, node := range ns.Items {
+		if _, ok := node.Labels[masterRoleLabel]; ok {
+			masterCount++
+		}
+		if _, ok := node.Labels[workerRoleLabel]; ok {
+			workerCount++
+		}
+	}
+
+	mon.emitGauge("cluster.summary", 1, map[string]string{
+		"actualVersion":     actualVersion(cv),
+		"desiredVersion":    desiredVersion(cv),
+		"masterCount":       strconv.Itoa(masterCount),
+		"workerCount":       strconv.Itoa(workerCount),
+		"provisioningState": mon.oc.Properties.ProvisioningState.String(),
+	})
```

Comment:
I am hesitant about this for a few reasons:
1. some items are duplicated elsewhere - this implies that cluster.summary could easily become a dumping ground.  What's wrong with having multiple graphs from different sources on a single dashboard page?  Why do we need to aggregate here?
2. provisioningState: note that our dashboards implicitly do not include clusters in state Creating, Failed/Creating, Deleting or Failed/Deleting.  Does that fit with the actual requirement (whatever that is?)?

Reply:
1. The main challenge is that we need an easy way to get table based metric/summary. With multiple metrics, somebody has to go in, get multiple metrics dumps, and play "data scientists" in the excel spreadsheet. I'm trying to avoid this work. Or we need to start doing join metrics in geneva, which I don't think is possible (and if it is it just moves the complexity to the proprietary system). 
The goal is to have better reporting capability. 
2. Same as 1. The goal is to see our estate state. 

Cluster summary is expected to grow. If you think of a better way to do reporting which do not involve manual work (something in next month, not half year) - I'm open to suggestions.